### DISCUSSION TITLE
Updated App Tracking Transparency(ATT) framework and GMA SDK

### DISCUSSION THREAD

**Discussion by Gokulv617** [airsdk/Adobe-Runtime-Support#812] [Q&A] [answered]
My Google Strategic Partnerships Manager is warning me on the following things

> iOS 14.5 is launching on 26th April and you will see a decline in revenue going forward. In case you still haven't updated App Tracking Transparency(ATT) framework and GMA SDK, the loss can be significant

What should I change according to this ?

**Comment by marchbold** [comment, marked as answer]
Hi,

Basically the change they are talking about limits the access of the advertising identifier. You now have to request authorisation to access it. Which means any Ad SDK that uses the id to track users and target adverts to them will be affected, **potentially** reducing the return you get from the advertising. 

If you want to integrate it you can use the IDFA ANE (https://github.com/distriqt/ANE-IDFA) to request authorisation:

https://github.com/distriqt/ANE-IDFA/wiki/u.Get-Advertising-Identifier#authorisation

**Reply by P2b10** [reply]
Hi marchbold,

If I am using your Adverts ANE is enough for this issue? or do I have to include the IDFA ANE?

**Reply by marchbold** [reply]
Depends, AdMob have included some ability to display the dialog through their UMP implementation, however in our apps we still use IDFA to directly access the iOS dialog to control exactly where it is presented and to understand adoption etc.

**Reply by GeneralVimes** [reply]
@marchbold thank you so much for the IDFA ANE!
I tried to add it into my game, but got a packaging error. The error persists even on an empty projects with only IDFA ANE. Maybe it has something to do with the newest AIR version (33.1.1.533)?

Here is how the error looks like:
Undefined symbols for architecture armv7:
  "_OBJC_CLASS_$_DTFREUtils", referenced from:
      objc-class-ref in <my user folder>\AppData\Local\Temp\fcf32071-4e83-4bb2-b872-9017efca1c9a\libcom.distriqt.IDFA.a(DTIDFAAIRExtensionContext.o)
      objc-class-ref in <my user folder>\AppData\Local\Temp\fcf32071-4e83-4bb2-b872-9017efca1c9a\libcom.distriqt.IDFA.a(IDFA.o)
ld: symbol(s) not found for architecture armv7
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_DTFREUtils", referenced from:
      objc-class-ref in <my user folder>\AppData\Local\Temp\fcf32071-4e83-4bb2-b872-9017efca1c9a\libcom.distriqt.IDFA.a(DTIDFAAIRExtensionContext.o)
      objc-class-ref in <my user folder>\AppData\Local\Temp\fcf32071-4e83-4bb2-b872-9017efca1c9a\libcom.distriqt.IDFA.a(IDFA.o)
ld: symbol(s) not found for architecture arm64
Compilation failed while executing : ld64

Is there a way to fix it, please? I'll be happy to run more tests, if needed.

**Reply by GeneralVimes** [reply]
Oh, wait, I did not read that The Core ANE. Let me try once more...

**Reply by GeneralVimes** [reply]
It works perfectly!
Thank you so much!

**Reply by marchbold** [reply]
Glad you found it, that `DTFREUtils` missing reference is in our Core ANE so yeah you were just missing that. Cheers.
